.github/workflows/file-tester.yml

```diff
@@ -7,12 +7,12 @@ name: File Tester
 on:
   pull_request:
     paths:
-      - 'utils/file-tester/**.py'
+      - 'utils/file-tester/**'
+      - '!utils/file-tester/**/*.md'
   push:
-    branches:
-      - main
     paths:
-      - 'utils/file-tester/**.py'

```

Comment:
Note: it seems that you can't combine branches & paths, so if we leave `branches` in then every push to `main` will trigger execution of every job.